### PR TITLE
Ensure combineSpectra works also with a non-empty processing queue

### PR DESCRIPTION
**Pull request by jorainer**
This PR fixes `combineSpectra` on a `Spectra` with a non-empty processing queue (issue #140).

### CODE REVIEW

File: R/Spectra-functions.R [hunk 1]
```diff
@@ -427,11 +428,14 @@ combineSpectra <- function(x, f = x$dataStorage, p = x$dataStorage,
     if (isReadOnly(x@backend))
         message("Backend of the input object is read-only, will change that",
                 " to an 'MsBackendDataFrame'")
-    ## We split the workload by storage file. This ensures memory efficiency
-    ## for file-based backends.
-    res <- bpmapply(FUN = .combine_spectra, split(x, p), split(f, p),
-                    MoreArgs = list(FUN = FUN, ...), BPPARAM = BPPARAM)
-    .concatenate_spectra(res)
+    if (length(levels(p)) > 1) {
```
Comment (sgibb) [inline R/Spectra-functions.R:431]: It is exactly the same code but I find it easier to read. You can ignore it if you like:
```suggestion
    if (nlevels(p)) > 1) {
```

File: R/Spectra.R [hunk 2]
```diff
@@ -389,8 +389,13 @@ NULL
 #'   split the input data for parallel processing.
 #'   This defaults to `p = x$dataStorage` and hence a per-file parallel
 #'   processing is applied for `Spectra` with file-based backends (such as the
-#'   [MsBackendMzR()]). The function returns a `Spectra` of length equal to the
-#'   unique levels of `f`.
+#'   [MsBackendMzR()]).
+#'   Prior combination of the spectra any processings queued in the lazy
```
Comment (sgibb) [inline R/Spectra.R:393]:
```suggestion
#'   Prior combination of the spectra all processings queued in the lazy
```

File: R/Spectra.R [hunk 3]
```diff
@@ -389,8 +389,13 @@ NULL
...
+#'   evaluation queue are applied. Be aware that calling `combineSpectra` on a
+#'   `Spectra` object with an `MsBackendHdf5Peaks` backend will
```
Comment (sgibb) [inline R/Spectra.R:395]: That is true for all backends with read and write support, isn't it? So maybe add a few words like: "Be aware ... with a backend that allows modifications, e.g. `MsBackendHdf5Peaks`, will **overwrite** ..."

File: NEWS.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Spectra 0.99
 
+## Changes in 0.99.8
+
+- `combineSpectra` applies data processing steps in the processing queue prior
```
Comment (sgibb) [inline NEWS.md:5]:
```suggestion
- `combineSpectra` applies data processing steps in the processing queue prior to
```

**Review by sgibb — APPROVED** [review]
Nice PR. I have just a few very minor suggestions. You could merge it with/without respecting them.

File: R/Spectra.R (hunk 3 above)
Comment (jorainer) [reply R/Spectra.R:395]: Thanks. I will add/update.

**Comment by jorainer** [comment]
Thanks @sgibb for the review! I will in addition also fix the issue/bug in Windows and merge.

**Pull request merged**